File: pulsar-third/pulsar-chrome-devtools/src/main/java/com/github/kklisura/cdt/protocol/commands/DeviceOrientation.java
package com.github.kklisura.cdt.protocol.commands;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2019 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.support.annotations.Experimental;
import com.github.kklisura.cdt.protocol.support.annotations.ParamName;

@Experimental
public interface DeviceOrientation {

  /** Clears the overridden Device Orientation. */
  void clearDeviceOrientationOverride();

  /**
   * Overrides the Device Orientation.
   *
   * @param alpha Mock alpha
   * @param beta Mock beta
   * @param gamma Mock gamma
   */
  void setDeviceOrientationOverride(
      @ParamName("alpha") Double alpha,
      @ParamName("beta") Double beta,
      @ParamName("gamma") Double gamma);
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/commands/DeviceOrientation.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.commands

import ai.platon.cdt.kt.protocol.support.annotations.Experimental
import ai.platon.cdt.kt.protocol.support.annotations.ParamName
import kotlin.Double

@Experimental
interface DeviceOrientation {
  /**
   * Clears the overridden Device Orientation.
   */
  suspend fun clearDeviceOrientationOverride()

  /**
   * Overrides the Device Orientation.
   * @param alpha Mock alpha
   * @param beta Mock beta
   * @param gamma Mock gamma
   */
  suspend fun setDeviceOrientationOverride(
    @ParamName("alpha") alpha: Double,
    @ParamName("beta") beta: Double,
    @ParamName("gamma") gamma: Double,
  )
}
